server/src/main/java/keywhiz/service/resources/automation/AutomationSecretResource.java

```diff
@@ -128,7 +130,7 @@ public AutomationSecretResponse createSecret(
       throw new ConflictException(format("Cannot create secret %s.", request.name));
     }
     ImmutableList<Group> groups =
-        ImmutableList.copyOf(aclDAO.getGroupsFor(secret));
+        ImmutableList.copyOf(aclDAOReadOnly.getGroupsFor(secret));
```

Comment:
Yeah, it might see outdated groups. I can update this to use the R/W aclDAO instead.